### PR TITLE
Fire Elemental Death Effects

### PR DESCRIPTION
**Pull request by Wisemonster**
[QoL]

## About the PR 
This changes the fire elementals so that they turn to ash when they die, and play sound/impact_sounds/burn_sizzle.ogg as well.

## Why's this needed? 
Currently, the fire elemental kinda just sits there burning when they die, often causing players to continue attacking them given that the only real tell is a small chat message. Plus, the sprite sometimes rotates upside down after they die. This change aims to give them their own unique death effect, so that other players will know for sure they are dead.

### CODE REVIEW

File: code/mob/living/critter/arena/fire_elemental.dm [hunk 1]
```diff
@@ -4,7 +4,7 @@
 	desc = "Oh god."
 	density = 1
 	icon_state = "fire_elemental"
-	icon_state_dead = "fire_elemental-dead"
+	//icon_state_dead = "fire_elemental-dead" No longer needed with the new death effects
```
Comment (ZeWaka) [inline code/mob/living/critter/arena/fire_elemental.dm:7]: Then remove it

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/mob/living/critter/arena/fire_elemental.dm [hunk 2]
```diff
@@ -67,3 +66,13 @@
...
+	death(var/gibbed)
+		..(gibbed, 0)
+		if (!gibbed)
+			playsound(src.loc, "sound/impact_sounds/burn_sizzle.ogg", 100, 1)
+			make_cleanable(/obj/decal/cleanable/ash,src.loc)
+			ghostize()
+			qdel(src)
+		else
+			playsound(src.loc, "sound/impact_sounds/burn_sizzle.ogg", 100, 1)
+			make_cleanable(/obj/decal/cleanable/ash,src.loc)
```
Comment (ZeWaka) [inline code/mob/living/critter/arena/fire_elemental.dm:78]: This can be consolidated - the only difference is the ghost part.

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/mob/living/critter/arena/fire_elemental.dm (hunk 2 above)
Comment (Wisemonster) [reply code/mob/living/critter/arena/fire_elemental.dm:78]: Fair enough. I'll probably make a pr to clean up the gunbot critter antag code, considering I based this on that code.

File: code/mob/living/critter/arena/fire_elemental.dm [hunk 3]
```diff
@@ -67,3 +66,10 @@
 
 	get_disorient_protection_eye()
 		return(max(..(), 80))
+	death(var/gibbed)
```
Comment (ZeWaka) [inline code/mob/living/critter/arena/fire_elemental.dm:69]:
```suggestion
		return(max(..(), 80))

	death(var/gibbed)
```

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**